pkg/resources/kafka/kafka.go

```diff
@@ -290,6 +290,10 @@ func (r *Reconciler) Reconcile(log logr.Logger) error {
 		return err
 	}
 
+	if err = r.reconcileHeadlessService(log); err != nil {
```

Comment:
It's quite confusing to have  `r.headlessService()` invoked at the beginning and then`r.reconcileHeadlessService()`. The handling of the headless service (deletion, creation, update) should be done in a single place `r.headlessService()`

Reply:
Sorry for the late.

But in the reconsiling the old kafka services are still in use, so we should not delete it until the end of reconciling.

Reply:
you are right. Meanwhile I worked on this as I has some spare time to be able to get this fix in, please see: https://github.com/banzaicloud/koperator/pull/799